### ISSUE TITLE
Expose parse tree via an extension API

### ISSUE THREAD

**Issue by Gama11** [vshaxe/vshaxe#104]
(once it is ready for that, that is)

@wiggin77 has expressed strong interest in this, since he needs a proper parser to fix some issues in codedox:
- https://github.com/vshaxe/codedox/issues/7
- https://github.com/vshaxe/codedox/issues/9

**Comment by Gama11** [comment]
Instead of a hxparser tree, we now have a tokentree. It might still be useful to expose that via an extension API though. Maybe haxe-test-adapter could do some on-the-fly auto-discovery of tests that way for instance.

**Comment by wiggin77** [comment]
Any news on this availability?  I've got a feature request (on top of the long standing ones) that would require an ability to query a parsed tree.  Cheers.

**Comment by Gama11** [comment]
Are you talking about https://github.com/vshaxe/codedox/issues/27? That's not even Haxe. :)

**Comment by wiggin77** [comment]
> Are you talking about [vshaxe/codedox#27](https://github.com/vshaxe/codedox/issues/27)? That's not even Haxe. :)

True, :-).  However, it does remind me to circle back and see if I can move away from regex function parsing into something more robust.
